include/deal.II/numerics/fe_field_function.templates.h

```diff
@@ -140,7 +141,14 @@ namespace Functions
     FEValues<dim> fe_v(mapping, cell->get_fe(), quad,
                        update_gradients);
     fe_v.reinit(cell);
-    fe_v.get_function_gradients(data_vector, gradients);
+
+    // Unfortunately we still need a temporary argument as we want to
+    // evaluate a gradient of a (generally) multicomponent function at
+    // a single quadrature point.
+    std::vector< std::vector<Tensor<1,dim,number> > > vgrads
+    (1,  std::vector<Tensor<1,dim,number> >(n_components) );
+    fe_v.get_function_gradients(data_vector, vgrads);
+    gradients = std::vector<Tensor<1,dim,number> >(vgrads[0].begin(), vgrads[0].end());
```

Comment:
I don't understand this patch. What you are saying is that the previous call matches the correct argument list, but the exception I had shows that the size was wrong. This can only mean that the `gradients` variable had the wrong size going into this function. You paper over this by creating a temporary of the right size, and then assigning it to `gradients`. But I think the correct patch should be to find out *why* `gradients` had the wrong size.

Or are you saying that we now call the *wrong overload* of `FEValues::get_function_gradients()` after the previous changes?